File: moe/tools/moe.tools.common/src/main/java/org/moe/gradle/model/GradlePluginModel.java
package org.moe.gradle.model;

import org.jetbrains.annotations.NotNull;

import java.util.Collection;

/** Returns artifacts information of a module */
public interface GradlePluginModel {
    /**
     * Returns a list of applied plugins.
     */
    @NotNull
    Collection<String> getGradlePluginList();
}

File: moe/tools/moe.tools.common/src/main/kotlin/org/moe/gradle/model/GradlePluginModel.kt
package org.moe.gradle.model

/** Returns artifacts information of a module  */
interface GradlePluginModel {

    /**
     * The MOE gradle plugin version
     */
    val version: String?

    /**
     * Returns a list of applied plugins.
     */
    val gradlePluginList: Collection<String>

    /**
     * Returns SDK properties
     */
    val sdkProperties: MOESdkProperties?

    /**
     * Returns Xcode properties
     */
    val xcodeProperties: MOEXcodeProperties?
}
